src/app/pages/SpotTradingPage/components/SpotHistory/index.tsx

```diff
@@ -1,110 +1,66 @@
   const { t } = useTranslation();
   const assets = AssetsDictionary.list();
-  const retry = useTradeHistoryRetry();
 
-  const cancelTokenSource = useRef<CancelTokenSource>();
-  const currentlyLoading = useRef<boolean>(false);
+  const [page, setPage] = useState(1);
 
-  const getHistory = useCallback(() => {
-    currentlyLoading.current = true;
-    setLoading(true);
-    if (cancelTokenSource.current) {
-      cancelTokenSource.current.cancel();
-    }
-    cancelTokenSource.current = axios.CancelToken.source();
-    log('start loading');
-    axios
-      .get(`${url}/events/conversion-swap/${account}`, {
-        cancelToken: cancelTokenSource.current.token,
-      })
-      .then(res => {
-        log('data returned', res.data);
-        setHistory(
-          res.data
-            .sort(
-              (a, b) =>
-                new Date(b.timestamp).getTime() -
-                new Date(a.timestamp).getTime(),
-            )
-            .map(item => {
-              const { assetFrom, assetTo } = extractAssets(
-                item.from_token,
-                item.to_token,
-              );
-              const order = getOrder(assetFrom.asset, assetTo.asset);
-              if (!order) return null;
-
-              return {
-                assetFrom,
-                assetTo,
-                order,
-                item,
-              };
-            })
-            .filter(item => item),
+  const { data, loading } = useGetSwapHistoryQuery({
+    variables: { user: account.toLowerCase() },
+    pollInterval: APOLLO_POLL_INTERVAL,
+  });
+
+  const history = useMemo(() => {
+    if (loading || !data) return [];
+
+    return data.swaps
+      .map(item => {
+        const { assetFrom, assetTo } = extractAssets(
+          item.fromToken.id,
+          item.toToken.id,
         );
-        setLoading(false);
-      })
-      .catch(e => {
-        if (!axios.isCancel(e)) {
-          error('loading failed', e);
-          setHistory([]);
-          setCurrentHistory([]);
-          setLoading(false);
-        }
+
+        if (!assetFrom || !assetTo) return null;
+        return {
+          toAmount: item.toAmount,
+          fromAmount: item.fromAmount,
+          timestamp: item.transaction.timestamp,
+          transactionHash: item.transaction.id,
+          assetFrom,
+          assetTo,
+        };
       })
-      .finally(() => (currentlyLoading.current = false));
-  }, [account, error, log, url]);
-
-  //GET HISTORY
-  useEffect(() => {
-    // load only if it's not already loading
-    if (account && !currentlyLoading.current) {
-      getHistory();
-    }
-  }, [account, getHistory, log, retry]);
-
-  const onPageChanged = useCallback(
-    data => {
-      const { currentPage, pageLimit } = data;
-      const offset = (currentPage - 1) * pageLimit;
-      setCurrentHistory(history.slice(offset, offset + pageLimit));
-    },
-    [history],
+      .filter(item => item);
```

Comment:
What is this `filter` supposed to do? I don't think you need it.

Reply:
If assetFrom OR assetTo is not found from the token id then the item would be null and this filter would make sure to remove those items